name: Django CI

on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]

jobs:
  build:
    runs-on: ubuntu-latest

    services:
      postgres:
        image: postgres:13
        env:
          POSTGRES_USER: bookstoreebac_db_user
          POSTGRES_PASSWORD: ShYdIoEFCpKY3rOJ6crNhQgP0ktBJ2DP
          POSTGRES_DB: bookstoreebac_db
        ports:
          - 5432:5432
        options: >-
          --health-cmd pg_isready
          --health-interval 10s
          --health-timeout 5s
          --health-retries 5

    env:
      DATABASE_URL: postgres://bookstoreebac_db_user:ShYdIoEFCpKY3rOJ6crNhQgP0ktBJ2DP@localhost:5432/bookstoreebac_db
      DJANGO_SETTINGS_MODULE: bookstore.settings
      SECRET_KEY: django-insecure-^d$@g!%52gimar0=12+2l67(2f6psk4#m-_+&df*9tsfb9k0dy
      DEBUG: False

    steps:
    - uses: actions/checkout@v2

    - name: Set up Python 3.12
      uses: actions/setup-python@v2
      with:
        python-version: 3.12

    - name: Install dependencies
      run: |
        pip install poetry
        poetry install

    - name: Run migrations
      run: |
        poetry run python manage.py migrate

    - name: Run tests
      run: |
        poetry run python manage.py test
